src/Microsoft.Health.Fhir.Azure/ExportDestinationArtifactProvider.cs

```diff
@@ -0,0 +1,100 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient;
+
+namespace Microsoft.Health.Fhir.Azure
+{
+    public class ExportDestinationArtifactProvider : IArtifactProvider
+    {
+        private const string ANONYMIZATIONCONTAINER = "anonymization";
+
+        private IExportClientInitializer<CloudBlobClient> _exportClientInitializer;
+        private ExportJobConfiguration _exportJobConfiguration;
+        private CloudBlobClient _blobClient;
+
+        public ExportDestinationArtifactProvider(
+            IExportClientInitializer<CloudBlobClient> exportClientInitializer,
+            IOptions<ExportJobConfiguration> exportJobConfiguration)
+        {
+            _exportClientInitializer = exportClientInitializer;
```

Comment:
Check for nulls `EnsureArg.IsNotNull(exportClientInitializer, nameof(exportClientInitializer))`

Reply:
Thanks, updated.